command/connect/envoy/envoy.go

```diff
@@ -22,13 +23,14 @@ import (
 	"github.com/hashicorp/consul/tlsutil"
 )
 
-func New(ui cli.Ui) *cmd {
+func New(ui cli.Ui, osPlatform string) *cmd {
 	c := &cmd{UI: ui}
-	c.init()
+	c.init(osPlatform)
 	return c
 }
 
-const DefaultAdminAccessLogPath = "/dev/null"
+const DefaultUnixAdminAccessLogPath = "/dev/null"
+const DefaultWindowsAdminAccessLogPath = "nul"
```

Comment:
The general approach that the Consul team took is to have OS specific values defined in OS specific files. I think if we create a windows and a non windows files with each defining different value of `DefaultAdminAccessLogPath` it would reduce the number of branches in the code below.